### PR TITLE
Extract more mixin module to compat

### PR DESCRIPTION
**Pull request by cosmo0920**
Follows up https://github.com/fluent/fluentd/pull/1044.
I've extracted modules which are defined in `mixin.rb` into compat entirely.

### CODE REVIEW

File: lib/fluent/compat/set_time_key_mixin.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+           raise Fluent::ConfigError, "Invalid boolean expression '#{s}' for include_time_key parameter" if include_time_key.nil?
+
+           @include_time_key = include_time_key
+         end
+
+         if @include_time_key
+           @time_key     = conf['time_key'] || 'time'
+           @time_format  = conf['time_format']
+
+           if    conf['localtime']
```
Comment (tagomoris) [inline lib/fluent/compat/set_time_key_mixin.rb]: Additional spaces

**Comment by tagomoris** [comment]
Almost code looks good.

File: lib/fluent/compat/set_time_key_mixin.rb (hunk 1 above)
Comment (cosmo0920) [inline lib/fluent/compat/set_time_key_mixin.rb]: I guess these additional spaces to clarify getting localtime/utc information from conf object.
Former is to get `localtime`, latter is to get `utc` from conf object.

Comment (tagomoris) [inline lib/fluent/compat/set_time_key_mixin.rb]: Ah... Okay, understood.

**Comment by tagomoris** [comment]
I'll merge this later.

**Comment by tagomoris** [comment]
Merged. Thank you!

**Pull request merged**